### PR TITLE
Add proofs for bitwise lemmas in dafny

### PR DESCRIPTION
**Pull request by jcp19**

### CODE REVIEW

File: verification/utils/bitwise/proofs.dfy [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+lemma BitAnd3_32bit(b: bv32)
+    ensures var res := b & 0x3;
+        0 <= res && res <= 3  &&
+        (b == 0 ==> res == 0) &&
+        (b == 3 ==> res == 3) &&
+        (b == 4 ==> res == 0)
+{}
+
+lemma BitAnd3_64bit(b: bv32)
```
Comment (Dspil) [inline verification/utils/bitwise/proofs.dfy:23]: How are those two different?

File: verification/utils/bitwise/proofs.dfy [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        (b == 3 ==> res == 3) &&
+        (b == 4 ==> res == 0)
+{}
+
+lemma BitAnd7_32bit(b: bv32)
+    ensures var res := b & 0x7;
+        0 <= res <= 7
+{}
+
+lemma BitAnd7_64bit(b: bv32)
```
Comment (Dspil) [inline verification/utils/bitwise/proofs.dfy:36]: same. shouldn't b here `b` be bv64 in the second case?

**Review by Dspil — APPROVED** [review]

File: verification/utils/bitwise/proofs.dfy (hunk 1 above)
Comment (jcp19) [inline verification/utils/bitwise/proofs.dfy:23]:
```suggestion
lemma BitAnd3_64bit(b: bv64)
```

File: verification/utils/bitwise/proofs.dfy (hunk 2 above)
Comment (jcp19) [inline verification/utils/bitwise/proofs.dfy:36]:
```suggestion
lemma BitAnd7_64bit(b: bv64)
```

Comment (jcp19) [reply verification/utils/bitwise/proofs.dfy]: Good catch! I have fixed both cases

**Pull request merged**